nebula-unity/Assets/scripts/GameManager.cs

```diff
@@ -18,9 +18,81 @@ public class GameManager : MonoBehaviour {
 	}
 
 	// TODO - Sterilize Input
-	public void GenerateNewMap(int size = 50){
-		if (map != null) { Destroy (map.gameObject); }
+	public void GenerateNewMap(){
+        if (map != null) { Destroy (map.gameObject); }
 
-		map = generatorScript.GenerateMap (size, size);
-	}
+        map = MakeMeAMap ();
+    }
+
+	/*
+    =
+    =
+    =
+    EXPOSE GENERATOR VARIABLES
+    =
+    =
+    =
+    */
+    int[] SetRoomDiameterRange(){
+		int minRoomDiameter = 1;
+        if (GameObject.Find("MinRoomDiameter").GetComponent<Text>().text == "" || minRoomDiameter == 3)
+            minRoomDiameter = -1;
+        else
+            minRoomDiameter = int.Parse(GameObject.Find("MinRoomDiameter").GetComponent<Text>().text);
+        if (minRoomDiameter < 3)
+            minRoomDiameter = 3;
+
+		int maxRoomDiameter;
+        if (GameObject.Find("MaxRoomDiameter").GetComponent<Text>().text == "")
+            maxRoomDiameter = -1;
+        else
+            maxRoomDiameter = int.Parse(GameObject.Find("MaxRoomDiameter").GetComponent<Text>().text);
+        if (maxRoomDiameter > 20)
+            maxRoomDiameter = 20;
+        if (minRoomDiameter > maxRoomDiameter){
+            int temp = minRoomDiameter;
+            minRoomDiameter = maxRoomDiameter;
+            maxRoomDiameter = temp;
+        }
+
+		int[] range = {minRoomDiameter, maxRoomDiameter};
+		return range;
+    }
+    float SetShapeFactor(){
+		float shapeFactor;
+        if (GameObject.Find("ShapeFactor").GetComponent<Text>().text == "")
+            shapeFactor = -1f;
+        else
+            shapeFactor = float.Parse(GameObject.Find("ShapeFactor").GetComponent<Text>().text);
+        if (shapeFactor > 10f)
+            shapeFactor = 10f;
+		return shapeFactor;
+    }
+    int SetMaxAttempts(){
+		int maxAttempts;
+        if (GameObject.Find("MaxAttempts").GetComponent<Text>().text == "")
+            maxAttempts = -1;
+        else
+            maxAttempts = int.Parse(GameObject.Find("MaxAttempts").GetComponent<Text>().text);
+		return maxAttempts;
+    }
+    int SetSize(){
+        int tempsize;
+        if (GameObject.Find("Size").GetComponent<Text>().text != "")
+            tempsize = int.Parse(GameObject.Find("Size").GetComponent<Text>().text);
+        else
+            tempsize = -1;
+        if (tempsize < 50)
+            tempsize = 50;
```

Comment:
Just as above, lets stay away from hardcoded values!   Top-level var such as `public static int MIN_MAP_DIMENSION = 50;` would be nicer.   Additionally, if you are willing to do the extra work, I'd like to allow non-square maps.   So letting us specify an x and y separately (such as 100x500 to create a long, thin map) would be really nice.